src/sentry/api/serializers/models/integration.py

```diff
@@ -99,10 +100,15 @@ def serialize(self, obj, attrs, user, include_config=True):
             config_data = obj.config
             dynamic_display_information = None
         else:
-            # just doing this to avoid querying for an object we already have
-            installation._org_integration = obj
-            config_data = installation.get_config_data()
-            dynamic_display_information = installation.get_dynamic_display_information()
+            try:
+                # just doing this to avoid querying for an object we already have
+                installation._org_integration = obj
+                config_data = installation.get_config_data()
+                dynamic_display_information = installation.get_dynamic_display_information()
+            except ApiError:
+                # If there is an ApiError from our 3rd party integration providers, assume there is an problem with the configuration and set it to disabled.
+                integration.update({"status": "disabled"})
+                return integration
```

Comment:
@NisanthanNanthakumar If you exit here, then `configData` is going to be `undefined` which is going to violate the type here: https://github.com/getsentry/sentry/blob/8549c003cfefb10ba81fce387dc4d5e36abccafe/src/sentry/static/sentry/app/types/index.tsx#L1199